controllers/util/solr_util.go

```diff
@@ -327,6 +328,13 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 		}
 	}
 
+	// Default preStop hook
+	preStop := &corev1.Handler{
```

Comment:
Is there a reason you split this out here?

Reply:
Just for better readability, I guess.

Reply:
> This looks good to me! But I would like to see a test added 🙂

I was checking around and found that we already have a few tests on Lifecycle, i.e. https://github.com/apache/solr-operator/blob/main/controllers/solrcloud_controller_externaldns_test.go#L117, https://github.com/apache/solr-operator/blob/main/controllers/solrcloud_controller_test.go#L300, https://github.com/apache/solr-operator/blob/main/controllers/solrcloud_controller_test.go#L396, and other places.

Are there any other kinds of tests that you think it would make sense to add? Perhaps just a simple custom input, not based on variables?

Reply:
Yeah, if you look at https://github.com/apache/solr-operator/blob/main/controllers/solrcloud_controller_test.go#L174, you'll see we test basically all of the custom options there. Add the custom lifecycle, and test that the podSpec actually uses it.

Reply:
Make sense. I've cleaned up the tests of `preStop` hook that were based on ENV variable `SOLR_PORT` because we're passing different Lifecycle values and that won't work; in any case, we already have a test for ENV variables. 
Also, I have added the Lifecycle tests for SolrCloud and PromExporter STS/Deployment. 

Please have a look and let me know if I miss anything.

Reply:
The new tests look great, but why did you remove the old tests from the tests that don't specify a custom lifecycle?

We test those because the ports change between the tests, and we want to make sure the correct port is always used no matter the solrAddressibility options specified.

It looks like we are keeping the same default lifecycle, unless I'm missing something. So those tests wouldn't have to change, I don't think.

Reply:
oh, yes, I thought it doesn't make sense, but now I see that the point of these tests is to ensure the correctness of the used port. I have added them back.

Reply:
thank you for all the help, Houston.

Reply:
No worries, thanks for the great and thorough PR!